packages/fuels-core/src/json_abi.rs

```diff
@@ -421,6 +419,97 @@ impl ABIParser {
         Ok(Token::Array(result))
     }
 
+    /// Creates a tuple `Token` from an array of parameter types and a string of values.
+    /// I.e. it takes a string containing values "value_1, value_2, value_3" and an array
+    /// of `ParamType` containing the type of each value, in order:
+    /// [ParamType::<Type of value_1>, ParamType::<Type of value_2>, ParamType::<Type of value_3>]
+    /// And attempts to return a `Token::Tuple()` containing the inner types.
+    /// It works for nested/recursive tuples.
+    pub fn tokenize_tuple(&self, value: &str, params: &[ParamType]) -> Result<Token, Error> {
+        if !value.starts_with('(') || !value.ends_with(')') {
+            return Err(Error::InvalidData(
+                "tuple value string must start and end with round brackets".into(),
+            ));
+        }
+
+        if value.chars().count() == 2 {
+            return Ok(Token::Tuple(vec![]));
+        }
+
+        let mut result = vec![];
+        let mut nested = 0isize;
+        let mut ignore = false;
+        let mut last_item = 1;
+        let mut params_iter = params.iter();
+
+        for (pos, ch) in value.chars().enumerate() {
```

Comment:
Even if this algo is in several places in the code, I've never fully understood it. Could you please use this PR as an opportunity to document it briefly? The variables and process.

Reply:
I added the comments inside the `tokenize_struct`, `tokenize_array` and `tokenize_tuple` functions